components/renku_data_services/base_api/pagination.py

```diff
@@ -13,10 +13,17 @@ class PaginationRequest(NamedTuple):
     page: int
     per_page: int
 
+    def __post_init__(self):
+        if self.page > 2**63 - 1:
```

Comment:
I do not understand the need for this guard. 

In Python, I believe integers will always be promoted when necessary and can accommodate arbitrary size. If this limit is imposed elsewhere, shouldn't we use unsigned 64-bit int (i.e., limit is 2**64 -1) and not signed ints?

Reply:
This is a postgres limitation on the size of indexes, and we cannot control this type.

Reply:
This is for postgres. Out automated tests on the api frequently pick very large number to test for edge cases and hit the limits on postgres. But I think we may have got the limit wrong. And it is as you suggest. I will try to get that error to show up again and see.

Reply:
The limit is correct. I looked it up, Postgres is using signed integers for this type. You can ignore my comment.